roles/setup_pgpool2/tasks/pgpool2_setup.yml

```diff
@@ -96,94 +96,166 @@
-  file:
+  ansible.builtin.file:
     path: "{{ pgpool2_pid_file_name | dirname }}"
     state: directory
     owner: "{{ pgpool2_user }}"
     group: "{{ pgpool2_group }}"
     mode: 0700
   when: pgpool2_pid_file_name is defined
-  become: yes
+  become: true
 
 # Build the configuration file
 - name: Build configuration file {{ pgpool2_configuration_file }}
-  template:
+  ansible.builtin.template:
     src: "./templates/pgpool.conf.template"
     dest: "{{ pgpool2_configuration_file }}"
     owner: "{{ pgpool2_user }}"
     group: "{{ pgpool2_group }}"
     mode: 0600
   become: true
 
-- name: Create pgpoolII systemd directory
-  file:
-    path: "{{ pgpool2_systemd_unit_file | dirname }}"
+- name: Create pgpoolII systemd directory for PG
+  ansible.builtin.file:
+    path: "{{ pgpool2_systemd_config_file | dirname }}"
     state: directory
     owner: "root"
     group: "root"
     mode: 0700
   when:
     - pg_type == 'PG'
-  become: yes
+  become: true
+
+- name: Create pgpoolII systemd directory for EPAS Debian
+  ansible.builtin.file:
+    path: "{{ pgpool2_systemd_config_file | dirname }}"
+    state: directory
+    owner: "root"
+    group: "root"
+    mode: 0700
+  when:
+    - pg_type == 'EPAS'
+    - ansible_os_family == 'Debian'
+  become: true
 
 - name: Generate pgpoolII systemd unit file
-  template:
+  ansible.builtin.template:
     backup: yes
-    dest: "{{ pgpool2_systemd_unit_file }}"
+    dest: "{{ pgpool2_systemd_config_file }}"
     src: ./templates/pgpool-II.unit.conf.template
     mode: 0644
   when:
     - pg_type == 'PG'
-  become: yes
+  become: true
+
+# add systemd config file to ensure environment file is referenced correctly
+- name: Generate pgpoolII systemd unit file for EPAS Debian
+  ansible.builtin.template:
+    backup: yes
+    dest: "{{ pgpool2_systemd_config_file }}"
+    src: ./templates/pgpool.unit.debian.conf.template
+    mode: 0644
+  when:
+    - pg_type == 'EPAS'
+    - ansible_os_family == 'Debian'
+  become: true
+
+# ensure system can find PID file
+- name: Insert PID location for Debian
+  ansible.builtin.lineinfile:
+    path: "{{ pgpool2_environment_file }}"
+    line: PIDFILE={{ pgpool2_pid_file_name }}
+  when:
+    - ansible_os_family == 'Debian'
+  become: true
+
+# original pgpool2_start_script is not accurate for EPAS Debian
+- name: Generate pgpoolII start bash script for EPAS Debian
+  ansible.builtin.template:
+    backup: yes
+    dest: "{{ pgpool2_start_script }}"
+    src: ./templates/edb-pgpool.sh.template
+    mode: 0755
+  when:
+    - pg_type == 'EPAS'
+    - ansible_os_family == 'Debian'
+  become: true
+
+# make sure items in pgpool2_bin_path are executable without directly referencing pgpool2_bin_path in command
+- name: Add pgpool executables to {{ pgpool2_user }} $PATH from {{ pgpool2_bin_path }}
+  ansible.builtin.shell:
+    cmd: >-
+      cp -r {{ pgpool2_bin_path }}/. /bin/
+  when:
+    - pg_type == 'EPAS'
+    - ansible_os_family == 'Debian'
+  become: true
```

Comment:
What problem does this task try to solve? Making hard copies of the binaries is not recommended because it will lead to problems when updating the package if the new binaries are not copied into /bin.
Binaries' locations can be configured through the system unit file if needed.

Reply:
When executing commands like `pcp_node_count`, the pgpool2 bin `/usr/edb/pgpool4.3/bin` is not on the `enterprisedb` user's `$PATH` and the command cannot be run without explicitly calling the bin, for example, `/usr/edb/pgpool4.3/bin/pcp_node_count`.

Reply:
I don't think not having `pcp*` command available in the current PATH is an issue. Please remove this task, making hard copies of binaries is not recommended. Thank you!